blog-posts/getting-started-gatsby-next-remix.md

```diff
@@ -0,0 +1,306 @@
+      This is useful if you have a page with a lot of static pages because then not all pages have to be generated at built time, which makes the built much faster.
+    - `blocking` → Same procedure as for `fallback: true`, except that there is no fallback while the HTML is being generated. There are only very special uses cases in which this makes sense, such as AMP.
+
+Remember to name the file after the relevant key from the `params` object with square brackets, in our case `[path].tsx`.
+To load the data for a blogpost the function `getStaticProps` is needed. This function now receives `{params}`, which can be used to determine the path and the depending data. Finally, in the actual page component, the data can be accessed via props (here via `{postData}`).
+
+```tsx
+export async function getStaticPaths() {
+    //     paths: [
+    //         {params: {path: 'first-blogpost'}},
+    //         {params: {path: 'second-blogpost'}},
+    //         ...
+    //         {params: {path: 'fifth-blogpost'}}
+    //     ],
+    const paths = getAllPostPaths()
+
+    return {
+        paths,
+        fallback: false
+    }
+
+export async function getStaticProps({params}: { params: PostsData }) {
+    const postData = await getPostData(params.path)
+    return {
+        props: {
+            postData
+        }
+    }
+}
+
+export default function Post({postData}: { postData: PostData }) {
+    return (
+            <div dangerouslySetInnerHTML={{__html: postData.contentHtml}}/>
+    )
+}
+```
+
+
+> 💡 The object returned by `getStaticProps` can include other keys besides `props`, such as `revalidate`. `revalidate` is set to `false` by default, but can also contain the amount in seconds after which a page re-generation can occur. In this way, individual static pages can be changed at a certain interval after being built. This is called [Incremental Static Regeneration (ISR)](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration).
+
+
+#### **Remix**
+
+Remix supports only server side rendering. An example of this can be found in the section below.
+
+## Server Side Rendering (SSR)
+
+With server side rendering (SSR), the HTML is rebuilt for each request. All three frameworks support SSR.
+
+#### **Gatsby**
+
+Besides static site generation (SSG) you can also use server side rendering (SSR) for certain use cases in Gatsby. For this, the function `getServerData` must be built into a page, in which the data is requested from the server. This data can then be accessed in the actual page component with `serverData`. In the following the example from the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/rendering-options/using-server-side-rendering/) is considered:
+
+```tsx
+import * as React from "react"
+
+const SSRPage = ({ serverData }) => (
+  <main>
+    <h1>SSR Page with Dogs</h1>
+    <img alt="Happy dog" src={serverData.message} />
+  </main>
+)
+
+export default SSRPage
+
+export async function getServerData() {
```

Comment:
How would I merge static and dynamic data? Or is this mutual exclusive? Say one dataset I can read via graphql and then it might update later. Is this covered? Does graphql play any role with SSR anymore ?

Reply:
Those are questions I ask because of my interest. You can decide if there is some answer that you might want to include but it's not something I feel missing as you get the point of SSR.

Reply:
Yes, you can mix it up: https://www.gatsbyjs.com/docs/reference/rendering-options/server-side-rendering/#interplay-with-build-time-graphql-queries